src/main/scala/com/gu/octopusthrift/services/ArticleFinder.scala

```diff
@@ -1,32 +1,75 @@
 package com.gu.octopusthrift.services
 
 import com.gu.octopusthrift.models._
+import scala.util.{ Failure, Success, Try }
 
 object ArticleFinder extends Logging {
 
-  def findBodyText(bundle: OctopusBundle): Option[OctopusArticle] = {
-    val bodyTexts: Array[OctopusArticle] =
-      bundle.articles.filter(hasBodyTextForWebPublication)
-    getPriorityBodyText(bodyTexts)
+  private val BodyText = "Body Text"
+  private val PanelText = "Panel Text"
+  private val TabularText = "Tabular Text"
+  private val ForBoth = "b"
+  private val ForWeb = "w"
+  private val ForPrint = "p"
+
+  // Order of precedence for object types: body over panel over tabular
+  private val bodyTextObjects = List(BodyText, PanelText, TabularText)
+
+  // Sometimes, object types look like Body Text [Ruled] and we don't care about the part in brackets
+  private def cleanObjectType(objectType: String): String = objectType.split("\\[").head.trim
+
+  // Prefer articles that are labelled both, then web, then print
+  private def articlesForPreferredDestination(
```

Comment:
Is there any particular reason that both is preferred to web only? Perhaps it should be the case that we only care about the fact it's destined for web and are not worried about if it's both or not.

Reply:
I'll double-check with @blishen and @hoyla as I thought this was something we'd discussed, but I can't find it documented anywhere.

Reply:
For posterity, we agreed after discussing with Blish to keep the filtering/sorting as it is in this PR, and review again when we have experience working with the data in Composer.